Document/0x06i-Testing-Code-Quality-and-Build-Settings.md

```diff
@@ -267,7 +267,7 @@ Although Xcode enables all binary security features by default, it may be releva
 
 -	**ARC** - Automatic Reference Counting - memory management feature
 	-	adds retain and release messages when required
--	**Stack Canary** - helps prevent buffer overflow attacks
+-	**Stack Canary** - helps prevent buffer overflow attacks by means of having a small integer right before the return pointer. A buffer overflow attack often overwrites a region of memory in order to overwrite the return pointer and take over the process-control. In that case, the canary gets overwritten as well. Tehrefore, the value of the cnaary is always checked to make sure it has not changed before a routine uses the return pointer on the stack.
```

Comment:
Tehrefore, the value of the cnaary -> Therefore, the value of th canary

Reply:
> Tehrefore, the value of the cnaary -> Therefore, the value of th canary
*the* canary